critwall/src/main/java/org/opencadc/critwall/BucketSelector.java

```diff
@@ -67,18 +67,83 @@
 
 package org.opencadc.critwall;
 
+import ca.nrc.cadc.util.StringUtil;
+import java.util.Iterator;
+import java.util.TreeSet;
 import org.apache.log4j.Logger;
+import org.opencadc.inventory.InventoryUtil;
+
 
-/**
- *
- * @author pdowler
- */
 public class BucketSelector {
-    private final Logger log = Logger.getLogger(BucketSelector.class);
-    public final String bucketSelector;
+    private static final Logger log = Logger.getLogger(BucketSelector.class);
+    // range as entered in .properties file
```

Comment:
First thing: minimise member variables
-  in this impl you only need the bucketList once the ctor is done, so that should be final; you do not need to keep bucketSelectors, rangeMin,  and rangeMax (*unless*: classes with state should override toString to assist debugging in FileSync and that should output "BucketSelector[" + rangeMin + "," + rangeMax +"]" so that could be a reason to keep those two member variables)
- maybe checkstyle is not enforcing this but static declarations are supposed to come before non-static and it helps comprehension to put them first

Just to make the limitations if this impl clear, I would declare a constant 
`private static final int MAX_PREFIX_LENGTH = 1;` 
to use in the ctor validation. 

I am also considering that the HEXVALUES constant should be declared in Artifact rather than here. If `public static final String Artifact.URI_BUCKET_CHARS` had been defined there, with a name like that, would it has made this whole thing easier to grok?

Reply:
Oh yeah.. if HEXVALUES had been in Artifact, it could have steered understanding. If it's likely it will be needed by other apps in the storage-inventory corral, having it central like that would be helpful.

Reply:
Based on what you have there, I'll keep rangeMin & rangeMax.